src/components/WorkOrder/MobileWorkingWorkOrderView.js

```diff
@@ -191,26 +196,50 @@ const MobileWorkingWorkOrderView = ({ workOrderReference }) => {
     )
 
     try {
-      if (files.length > 0) {
-        const description = data.description
-
-        const uploadResult = await uploadFiles(
-          files,
-          workOrderReference,
-          'Closing work order',
-          description,
-          (value) => setLoadingStatus(value)
+      if (workOrderFiles.length > 0 || followOnFiles.length > 0) {
+        // initiate both uploads
+        var totalFilesToUpload = workOrderFiles.length + followOnFiles.length
+
+        const fileUploadCompleteCallback = fileUploadStatusLogger(
+          totalFilesToUpload,
+          setLoadingStatus
         )
 
-        if (!uploadResult.success) {
-          setError(uploadResult.requestError)
-          setLoadingStatus(null)
-          return
+        if (workOrderFiles.length > 0) {
+          const uploadResult = await uploadFiles(
+            workOrderFiles,
+            workOrderReference,
+            'Closing work order',
+            data.workOrderPhotoDescription,
+            fileUploadCompleteCallback
+          )
+
+          if (!uploadResult.success) {
+            setError(uploadResult.requestError)
+            setLoadingStatus(null)
+            return
+          }
         }
 
-        setLoadingStatus('Completing workorder')
+        if (followOnFiles.length > 0) {
+          const uploadResult = await uploadFiles(
+            followOnFiles,
+            workOrderReference,
+            'Raising a follow on',
+            data.followOnPhotoDescription,
+            fileUploadCompleteCallback
+          )
+
+          if (!uploadResult.success) {
+            setError(uploadResult.requestError)
+            setLoadingStatus(null)
+            return
+          }
+        }
       }
```

Comment:
Combining these to run simultaneously may be required if performance becomes an issue